pcdsdevices/epics_motor.py

```diff
@@ -38,6 +39,30 @@
 logger = logging.getLogger(__name__)
 
 
+class MstaEnum(Enum):
+    """
+    Enum for the EPICS motor record .MSTA field bits.
+    """
+    # Note: the motor record docs start bit numbering at 1, but the MSTA bits
+    # start at 0.
+    direction = 0       # last raw move direction (0: negative, 1: positive)
+    done = 1            # motion is complete
+    plus_ls = 2         # plus limit switch is hit
+    home_ls = 3         # state of the home limit switch
+    # Bit 4 is un-used
```

Comment:
gotta love EPICS.  Is this a bitmask thing?

Reply:
Yep. The field and bit descriptions are in a table here: https://epics.anl.gov/bcda/synApps/motor/R5-6/motorRecord.html#Fields_status

Reply:
That doc says bit 6 is also unused.  Is our motor record different?

Reply:
My bad. That's an old version of the documentation. Here's the latest, which I believe matches what I have here: https://epics.anl.gov/bcda/synApps/motor/motorRecord.html#Fields_status

Reply:
Please make sure this work as expected. I seem to recall ( but am not 100% positive) that our very special IMS2 record as redefined the MSTA fields.

Reply:
Similarly, I'd suggest to also take a look at the Newport motor record.

Reply:
@silkenelson You're right. The IMS record and the Newport record does redefine some of these bits. I didn't see anything odd in the MCS2 motor record. I'll look into how to deal with this on Monday. 

Vanilla Motor Record: 

![image](https://github.com/user-attachments/assets/8fc58902-5d4a-4f36-a9ef-655a804f77a7)

Newport Motor Record: 

![image](https://github.com/user-attachments/assets/304328f6-951b-4aa3-9ca9-1d28c7cca40c)

Latest IMS motor record:

![Screenshot from 2024-07-21 09-34-46](https://github.com/user-attachments/assets/2024fe8f-3f94-40e8-a2b4-76bc9684a9d7)

Reply:
I think the most natural way to deal with this is to make different `MstaEnum`'s for different motor records.  `EpicsMotorInterface` might use the default one, then `pcdsdevices.epics_motor.Newport` gets its own, etc.

This is assuming we motor subclasses for these records.

Reply:
It looks like we have the following `EpicsMotorInterface` subclasses to consider from the `pcdsdevices` side of things.  (Of course we could have motors that don't have their own subclass)

A quick (turned not-so-quick) survey of the `EpicsMotorInterface` family tree

* `pcdsdevices.ccm.CCMPico`
* `pcdsdevices.epics_motor.PCDSMotorBase`
  * `pcdsdevices.epics_motor.IMS`
  * `pcdsdevices.epics_motor.Newport`
  * `pcdsdevices.epics_motor.DelayNewport` 
    * `pcdsdevices.lxe.TimeToolDelay`
    * `pcdsdevices.lxe._ReversedTimeToolDelay`
  * `pcdsdevices.epics_motor.PMC100`
  * `pcdsdevices.epics_motor.MMC100`
* `pcdsdevices.epics_motor.BeckhoffAxis`
  * `pcdsdevices.epics_motor.BechoffAxisEPS`
  * `pcdsdevices.epics_motor.BechoffAxisEPSCustom`
  * `pcdsdevices.epics_motor.BechoffAxis_Pre140` 
  * `pcdsdevices.epics_motor.BechoffAxisNoOffset`
* `pcdsdevices.epics_motor.SmarAct`


An interesting mapping we have:
```
    motor_types = (('MMS', IMS),
                   ('CLZ', IMS),
                   ('CLF', IMS),
                   ('MMN', Newport),
                   ('MZM', PMC100),
                   ('MMC', MMC100),
                   ('MMB', BeckhoffAxis),
                   ('PIC', PCDSMotorBase),
                   ('MCS', SmarAct),
                   ('MCS2', SmarAct))
```

Reply:
> I think the most natural way to deal with this is to make different `MstaEnum`'s for different motor records. `EpicsMotorInterface` might use the default one, then `pcdsdevices.epics_motor.Newport` gets its own, etc.
> 
> This is assuming we motor subclasses for these records.

I agree. This is the direction I've taken in my dev branch, just waiting to test.